src/backends/utils/cacheWasm.js

```diff
@@ -0,0 +1,83 @@
+import { getCache } from '../../utils/cache.js';
+
+/**
+ * Loads and caches a file from the given URL.
+ * @param {string} url The URL of the file to load.
+ * @returns {Promise<Response|import('../../utils/hub/FileResponse.js').default|null|string>} The response object, or null if loading failed.
+ */
+async function loadAndCacheFile(url) {
+    const fileName = url.split('/').pop();
+    try {
+        const cache = await getCache();
+
+        // Try to get from cache first
+        if (cache) {
+            try {
+                return await cache.match(url);
+            } catch (e) {
+                console.warn(`Error reading ${fileName} from cache:`, e);
```

Comment:
if `await cache.match(url)` returns undefined (i.e., it is not in cache), then we return `undefined` from this function... and the fetch below is never called (meaning, it is never cached).